featuregates/controller/pkg/feature/feature_suite_test.go

```diff
@@ -0,0 +1,553 @@
+package feature
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"math/big"
+	mrand "math/rand"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/apimachinery/pkg/types"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	corev1alpha2 "github.com/vmware-tanzu/tanzu-framework/apis/core/v1alpha2"
+)
+
+func TestFeaturegate(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Featuregate Suite")
+}
+
+var (
+	cfg       *rest.Config
+	k8sClient client.Client
+	testEnv   *envtest.Environment
+	ctx       context.Context
+	cancel    context.CancelFunc
+	setupLog  = ctrl.Log.WithName("controllers").WithName("FeatureGates")
+
+	timeout  = 5 * time.Second
+	interval = 100 * time.Millisecond
+)
+
+func generateCertificateAndManifests() error {
```

Comment:
Are there any utilities to create certs? Addons module does something similar, so you might want to look into it.

Reply:
The util used by addons is in addons module. If we import, we will be adding a go mod dependency on addon module and that is not necessary. I think what we are doing here to create the certificates are straightforward and no need of using utils from other modules. Let me know if there are other approaches.

Reply:
I think it's better we don't import stuff from addons module as it would be evicted from t-f soon, maybe we can move this code to [util module](https://github.com/vmware-tanzu/tanzu-framework/tree/main/util) for reusability purposes in other controllers

Reply:
Done